src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1248,5 +1248,176 @@ private static bool HasNamespaceName(NamespaceSymbol @namespace, string namespac
             var name = @namespace.Name;
             return (name.Length == length) && (string.Compare(name, 0, namespaceName, offset, length, comparison) == 0);
         }
+
+        internal static bool IsNonGenericTaskType(this TypeSymbol type, CSharpCompilation compilation)
+        {
+            var namedType = type as NamedTypeSymbol;
```

Comment:
I'd considered checking `Kind` but checking `as` instead avoids an additional check of `type != null`.